ScoutSuite/providers/gcp/resources/cloudsql/database_instances.py

```diff
@@ -0,0 +1,49 @@
+from ScoutSuite.providers.gcp.facade.gcp import GCPFacade
+from ScoutSuite.providers.gcp.resources.resources import GCPCompositeResources
+from ScoutSuite.providers.gcp.resources.cloudsql.backups import Backups
+from ScoutSuite.providers.gcp.resources.cloudsql.users import Users
+from ScoutSuite.providers.utils import get_non_provider_id
+
+class DatabaseInstances(GCPCompositeResources):
+    _children = [ 
+        (Backups, 'backups'),
+        (Users, 'users')
+    ]
+
+    def __init__(self, gcp_facade: GCPFacade, project_id: str):
+        self.gcp_facade = gcp_facade
+        self.project_id = project_id
+
+    async def fetch_all(self):
+        raw_instances = await self.gcp_facade.cloudsql.get_database_instances(self.project_id)
+        for raw_instance in raw_instances:
+            instance_id, instance = self._parse_instance(raw_instance)
+            self[instance_id] = instance
+            await self._fetch_children(self[instance_id], gcp_facade = self.gcp_facade, project_id = self.project_id, instance_name = instance['name'])
```

Comment:
I believe this should be parallelized using `asyncio.ensure_future`.

Reply:
Done!